### DISCUSSION TITLE
received_clone_create How do I get an entity?

### DISCUSSION THREAD

**Discussion by iwanttodo1t** [YimMenu/YimMenu#3365] [Q&A]
The hook provides the object id, is it possible to get the entity? I tried to get it, but it returned null, and I wonder if the entity is invincible and invisible

**Comment by Tayipc38** [comment]
`auto net_object = g_pointers->m_gta.m_get_net_object(*g_pointers->m_gta.m_network_object_mgr, object_id, true);
auto game_object = net_object->GetGameObject();
if(game_object)
{
auto entity_handle = g_pointers->m_gta.m_ptr_to_handle(game_object);
}`

Of the top of my head.

**Reply by iwanttodo1t** [reply]
Thamk you, I want to find out who generated the attack squad near me, I tried the code you provided and he returned null

**Reply by Tayipc38** [reply]
> Thamk you, I want to find out who generated the attack squad near me, I tried the code you provided and he returned null

I don't see how getting the entity script handle would yield any information as to who spawned a certain net_object. Besides, peds probably won't have the players id as their owner on your client.

**Reply by iwanttodo1t** [reply]
> > Thamk you, I want to find out who generated the attack squad near me, I tried the code you provided and he returned null
> 
> I don't see how getting the entity script handle would yield any information as to who spawned a certain net_object. Besides, peds probably won't have the players id as their owner on your client.

When another player spawns an attack team for me, received_clone_create will receive multiple messages containing the object type and object id and who spawned it for me（arg:src），I just tested it in a two person session, I'm not sure if the src is correct

**Reply by Tayipc38** [reply]
How will you differentiate what creation is part of an attack squad an what is legit?

**Reply by iwanttodo1t** [reply]
Invincibility and attacking my ped is illegal

**Reply by iwanttodo1t** [reply]
I have no idea about other situations

**Reply by iwanttodo1t** [reply]
I ended up having to handle this in the render_ped hook, which unfortunately doesn't get called if the attacking squad is invisible.
